swatch-producer-aws/src/main/java/com/redhat/swatch/util/ApiResponseHelper.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.redhat.swatch.util;
+
+import com.redhat.swatch.openapi.model.Errors;
+import java.util.Optional;
+import javax.ws.rs.core.Response;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class ApiResponseHelper {
```

Comment:
Should do this instead: https://quarkus.io/guides/rest-client-reactive#exception-handling

Reply:
@kahowell can you elaborate on what you mean here? Should I use a custom exception instead of APIException if response.getStatus == 404 and the code is SUBSCRIPTION_TERMINATED?

Reply:
What I mean is that we should configure the microprofile client to parse responses into specific exceptions, and that this would be a more idiomatic way to handle the conversion of responses into `Errors`. I'm a little surprised that the generated client doesn't already do this with the `Errors` schema... It might be worth digging a little in the openapi-generator templates to see if we can contribute a fix, but otherwise, it shouldn't be much effort to add a `ResponseExceptionMapper`. This requires a new exception class and a new `ResponseExceptionMapper`.